ouroboros-network/src/Ouroboros/Network/DeltaQ.hs

```diff
@@ -269,3 +275,21 @@ gsvRequestResponseDuration PeerGSV{outboundGSV, inboundGSV}
         gsvTrailingEdgeArrive outboundGSV reqSize
      <> gsvTrailingEdgeArrive inboundGSV respSize
 
+
+defaultGSV :: PeerGSV
+defaultGSV = PeerGSV { outboundGSV, inboundGSV }
+  where
+    default_g = 3600 -- 1h, start with an unreasonable large G.
```

Comment:
This needs to be `min (div maxBound 4, 3600)` so that we don't get bitten by 32ness later

Reply:
There isn't a maxBound for DiffTime.

Reply:
That's the problem, you can still declare to large `DiffTime` which will not fit into `Int32`. `threadDelay` is safe but `registerDelay` is not.

Reply:
ref: #2184

Reply:
This is changed to use to old hard coded g value (500ms). The reason for that is that we have to have a reasonable value in case the `KeepAlive` protocol isn't supported.